src/pages/CreatePool/index.tsx

```diff
@@ -110,34 +142,58 @@ export default function CreatePool(): React.ReactElement {
     }
   }
 
-  const poolNameError = poolName.length > 10
-  const poolSymbolError = poolSymbol.length > 14
-  const parameterError =
-    !isNumber(parameter) ||
-    parseFloat(parameter) < 1 ||
-    parseFloat(parameter) > 400
+  const poolNameError = poolName.length > 31
+  const poolSymbolError = poolSymbol.length > 31
+  const aParameterError = !isNumber(aParameter) || parseFloat(aParameter) < 1
```

Comment:
I just looked at the `isNumber` function and there are several gaps. Most importantly, aParameter can't be a decimal, so we don't need any parseFloat logic etc. Also the `isNumber` regex itself returns true for `""`. You probably just want `/^\d+$/` and a `> 0` check

Reply:
also `regex.test` instead of `.exec` @nkrulikovsky

Reply:
> I just looked at the `isNumber` function and there are several gaps. Most importantly, aParameter can't be a decimal, so we don't need any parseFloat logic etc. Also the `isNumber` regex itself returns true for `""`. You probably just want `/^\d+$/` and a `> 0` check

The regex you suggested doesn't quite work right. It shows an error by default for the input field with that regex. I changed the exec to test though. The parseFloat is casting the string to a number to validate it's not less than 1. The value being passed into the contract is a BigNumber from that string.